### PR TITLE
Add support for prompt roles and prompt preambles

### PR DESCRIPTION
**Pull request by ahejlsberg**
This PR adds support for prompt roles, i.e. prompt sections tagged with `"system"`, `"user"`, or `"assistant"` roles. TypeChat uses the `"user"` role for prompts it generates and the `"assistant"` role for previous LLM responses (which will be part of the prompt in repair attempts). TypeChat currently doesn't use the `"system"` role.

This PR also adds support for specifying a prompt preamble when processing requests. This makes it easier to implement chat-like interactions and conversation history.

### CODE REVIEW

File: src/typechat.ts [hunk 1]
```diff
@@ -111,7 +114,8 @@ export function createJsonTranslator<T extends object>(model: TypeChatLanguageMo
             if (!attemptRepair) {
                 return error(`JSON validation failed: ${validation.message}\n${jsonText}`);
             }
-            prompt += `${responseText}\n${typeChat.createRepairPrompt(validation.message)}`;
```
Comment (umeshma) [inline src/typechat.ts:114]: We'll be sending the responseText twice. Once as {assistant, responseText} and also as part of the prompt.

**Review by DanielRosenwasser — APPROVED** [review]

**Review by jonparker — APPROVED** [review]

**Comment by Githamza** [comment]
How can I try it ?  I have installed the last version (v0.0.10) of typechat but in node_modules/typechat, these additions does not exists

**Comment by Githamza** [comment]
Thanks , I have a question , why is the `request` parameter mandatory ? If I would send a promptPreamble with `system` and `user` roles , what should I put in `request` param ? for me `request` content is the `user` role content

**Comment by olawalejuwonm** [comment]
How can this be used please?

**Comment by kad3nce** [comment]
@olawalejuwonm, they key change is here: https://github.com/microsoft/TypeChat/pull/114/files#diff-1ece2a0b089b56f4aa22ef64be65b4305b5e890b560a39d9d12140f9a42d40f4R104

Add a `user` message preamble:
```
translator.translate(data, 'Hello there');
```

Add a system prompt:
```
translator.translate(data, [
  { role: 'system', content: 'You are an expert data anonymizer...' }
]);
```

**Comment by olawalejuwonm** [comment]
How do I pass the user content then?

**Pull request merged**